### PR TITLE
kafkauser: copy ca to secret

### PR DESCRIPTION
**Pull request by LuciferInLove**
## Description

* Copy istio CA from the cert-manager ClusterIssuer Secret to KafkaUser Secret
* Use `cert-manager-namespace` flag in both `certmanagerpki` and `k8scsrpki` packages

## Type of Change
- [x] New Feature

## Checklist
- [x] I have read the [contributing guidelines](../CONTRIBUTING.md)
- [x] Existing issues have been referenced (where applicable)
- [x] I have verified this change is not present in other open pull requests
- [x] Functionality is documented
- [x] All code style checks pass
- [x] All new and existing tests pass

### CODE REVIEW

File: pkg/pki/k8scsrpki/k8scsr_user.go
Comment (Kuvesz) [inline pkg/pki/k8scsrpki/k8scsr_user.go:353]: I think it may be worth having a variable instead of doing the string splitting twice. For me it would make the code look better, but it's very optional.

**Review by Kuvesz — APPROVED** [review]
LGTM, left a comment.

File: pkg/pki/k8scsrpki/k8scsr_user.go
Comment (LuciferInLove) [reply pkg/pki/k8scsrpki/k8scsr_user.go]: Thanks for noticing

**Review by Kuvesz — APPROVED** [review]

File: pkg/pki/k8scsrpki/k8scsr_user.go
Comment (panyuenlau) [inline pkg/pki/k8scsrpki/k8scsr_user.go:380]: Question: would it be possible that the `tls.crt` is somehow not present in the secret and we should return an error here if this happens?

**Review by panyuenlau — APPROVED** [review]
Left one question, but overall lgtm

File: pkg/pki/k8scsrpki/k8scsr_user.go
Comment (LuciferInLove) [reply pkg/pki/k8scsrpki/k8scsr_user.go:380]: Since I get a standard cert-manager secret, this field should always exist there. But I'll add a check just in case, thanks.

**Review by panyuenlau — DISMISSED** [review]

**Review by pregnor — DISMISSED** [review]
I'm not a cert expert, @bartam1 is better in this stuff, found 1 thing, otherwise LGTM to me

File: pkg/pki/k8scsrpki/k8scsr_user.go
Comment (LuciferInLove) [reply pkg/pki/k8scsrpki/k8scsr_user.go:356]: Even if the signer name is `{{signingReq.Spec.SignerName}}/` by accident, `getCAChain` won't be triggered because `cert-manager` can't issue a certificate with a wrong signerName. So, I can't imagine the case.

Comment (pregnor) [reply pkg/pki/k8scsrpki/k8scsr_user.go:356]: Yeah but we are relying on the fact that some outside factor deems it unlikely code ever gets here.

But if it does get here because
A, cert-manager lets this through in the future or
B, we refactor something and this accidentally gets called
this will result in a panic that could easily have been avoided.

I would rather see this fixed before merging, but I can let this go if you think it should go in as is.

Comment (pregnor) [reply pkg/pki/k8scsrpki/k8scsr_user.go:356]: Whatever, I guess I can fix it once it's in if it only bothers me.

**Review by pregnor — DISMISSED** [review]

File: pkg/pki/k8scsrpki/k8scsr_user.go
Comment (bartam1) [inline pkg/pki/k8scsrpki/k8scsr_user.go:393]: LGTM! 
Just one question.
Is the  certManagerSecret.Data[defaultCertManagerIssuerSecretCertificateFile] contains the leaf certificate? If yes then we should also include the leaf certificate when not cert-manager is used for k8s CSR signing.

**Review by bartam1 — APPROVED** [review]
LGTM!

**Comment by pregnor** [comment]
IMO it's mergable but I didn't dare to merge because didn't want to cause any headaches to #1020

**Comment by LuciferInLove** [comment]
> IMO it's mergable but I didn't dare to merge because didn't want to cause any headaches to #1020

Such a long-suffering request... 😃
If we don't plan to move to cert-manager anymore, I can close this PR. Otherwise, @bartam1 can approve if it creates more mess for #1020 or not. As another option, I can merge it later.

**Pull request merged**
